common/djangoapps/student/views/management.py

```diff
@@ -558,6 +576,11 @@ def activate_account(request, key):
         url_path = f'/login?account_activation_status={activation_message_type}'
         return redirect(settings.AUTHN_MICROFRONTEND_URL + url_path)
 
+    redirect_to, root_url = get_next_url_for_login_page(request, include_host=True)
+    if redirect_to:
```

Comment:
I guess you need to move this before the Authn MFE redirection check at L575? Otherwise, it will redirect the user to MFE if not authenticated.

Reply:
👍 Thank you for catching that.